scripts/snapshot.ts

```diff
@@ -0,0 +1,163 @@
+import Contracts from '../components/Contracts';
+import { getBalance } from '../test/helpers/Utils';
+import { DeployedContracts, isTenderlyFork } from '../utils/Deploy';
+import { NATIVE_TOKEN_ADDRESS } from '../utils/TokenData';
+import '@nomiclabs/hardhat-ethers';
+import '@typechain/hardhat';
+import Decimal from 'decimal.js';
+import { BigNumber } from 'ethers';
+import fs from 'fs';
+import { ethers, getNamedAccounts } from 'hardhat';
+import 'hardhat-deploy';
+
+interface PoolStats {
+    symbol: string;
+    bntTradingLiquidity: BigNumber;
+    tknTradingLiquidity: BigNumber;
+    stakedBalance: BigNumber;
+    depositLimit: BigNumber;
+    tradingFeePPM: number;
+    depositingEnabled: boolean;
+    tradingEnabled: boolean;
+    spotRate: Decimal;
+    emaRate: Decimal;
+    emaBlockNumber: number;
+    emaCompressedNumerator: BigNumber;
+    emaCompressedDenominator: BigNumber;
+    masterVaultTknBalance: BigNumber;
+    externalProtectionVaultTknBalance: BigNumber;
+    bntFundingLimit: BigNumber;
+    bntRemainingFunding: BigNumber;
+    bntFundingAmount: BigNumber;
+    bnTkn: string;
+    bnTknTotalSupply: BigNumber;
+}
+
+interface GlobalStats {
+    blockNumber: number;
+    masterVaultBntBalance: BigNumber;
+    bntPoolBnBntBalance: BigNumber;
+    bntPoolStakedAmount: BigNumber;
+    vortexBntBalance: BigNumber;
+    networkFeePPM: number;
+    minLiquidityForTrading: BigNumber;
+    lockDuration: number;
+    bnBntTotalSupply: BigNumber;
+}
+
+const gcd = (x: BigNumber, y: BigNumber) => {
+    while (y.gt(0)) {
+        const t = y;
+        y = x.mod(y);
+        x = t;
+    }
+
+    return x;
+};
+
+const saveCSV = (data: Record<string, any>[], outputPath: string) => {
+    const headers = Object.keys(data[0]);
+    const rows = data
+        .map((p) =>
+            Object.values(p)
+                .map((v) => v.toString())
```

Comment:
Using `toString()` on `Decimal` values may yield scientific-notation strings.
In order to avoid that, you can use `toFixed()` specifically for `Decimal` values.

Reply:
It should be ok for this case, since this CSV is being used in as a spreadsheet

Reply:
The problem is that once converted to scientific-notation, you lose data (i.e., the accuracy of the stored information is degraded).